modules/queue/base_levelqueue_unique.go

```diff
@@ -6,18 +6,21 @@ package queue
 import (
 	"context"
 	"sync"
-	"unsafe"
+	"sync/atomic"
 
 	"code.gitea.io/gitea/modules/nosql"
+	"code.gitea.io/gitea/modules/queue/lqinternal"
 
 	"gitea.com/lunny/levelqueue"
 	"github.com/syndtr/goleveldb/leveldb"
 )
 
 type baseLevelQueueUnique struct {
-	internal *levelqueue.UniqueQueue
-	conn     string
-	cfg      *BaseConfig
+	internal atomic.Pointer[levelqueue.UniqueQueue]
```

Comment:
Why must this be atomic now?

Reply:
Because `RemoveAll` needs to re-create the levelqueue. The levelqueue is used by different goroutines.